--- src/Orderbook.cpp.original	2025-12-14 21:00:00.000000000 +0530
+++ src/Orderbook.cpp	2025-12-14 21:01:00.000000000 +0530
@@ -13,8 +13,7 @@
 	{
 		const auto now = system_clock::now();
 		const auto now_c = system_clock::to_time_t(now);
-		std::tm now_parts;
-		localtime(&now_c, &now_parts);
+		std::tm now_parts = *std::localtime(&now_c);
 
 		if (now_parts.tm_hour >= end.count())
 			now_parts.tm_mday += 1;
@@ -39,10 +38,10 @@
 		{
 			std::scoped_lock ordersLock{ ordersMutex_ };
 
-			for (const auto& [_, entry] : orders_)
+			for (const auto& [orderId, entry] : orders_)
 			{
-				const auto& [order, _] = entry;
-
+				const auto& [order, location] = entry;
+
 				if (order->GetOrderType() != OrderType::GoodForDay)
 					continue;
 
